vyper/venom/passes/normalization.py

```diff
@@ -44,24 +49,34 @@ def _insert_split_basicblock(self, bb: IRBasicBlock, in_bb: IRBasicBlock) -> IRB
                     inst.operands[i] = split_bb.label
 
         # Update the labels in the data segment
-        for inst in self.ctx.data_segment:
+        for inst in fn.ctx.data_segment:
             if inst.opcode == "db" and inst.operands[0] == bb.label:
                 inst.operands[0] = split_bb.label
 
         return split_bb
 
-    def _run_pass(self, ctx: IRFunction) -> int:
-        self.ctx = ctx
+    def _run_pass(self) -> int:
+        fn = self.manager.function
         self.changes = 0
 
+        self.manager.request_analysis(CFGAnalysis)
+
         # Split blocks that need splitting
-        for bb in ctx.basic_blocks:
+        for bb in fn.basic_blocks:
             if len(bb.cfg_in) > 1:
                 self._split_basic_block(bb)
 
         # If we made changes, recalculate the cfg
         if self.changes > 0:
-            calculate_cfg(ctx)
-            ctx.remove_unreachable_blocks()
+            self.manager.force_analysis(CFGAnalysis)
+            fn.remove_unreachable_blocks()
```

Comment:
i think this is redundant, because `remove_unreachable_blocks()` also gets run in SimplifyCFG pass